File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/VendorLargeDataDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of vendor_large_data. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class VendorLargeDataDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final VendorLargeDataDbm _instance = new VendorLargeDataDbm();
    private VendorLargeDataDbm() {}
    public static VendorLargeDataDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((VendorLargeData)et).getLargeDataId(), (et, vl) -> ((VendorLargeData)et).setLargeDataId(ctl(vl)), "largeDataId");
        setupEpg(_epgMap, et -> ((VendorLargeData)et).getStringIndex(), (et, vl) -> ((VendorLargeData)et).setStringIndex((String)vl), "stringIndex");
        setupEpg(_epgMap, et -> ((VendorLargeData)et).getStringNoIndex(), (et, vl) -> ((VendorLargeData)et).setStringNoIndex((String)vl), "stringNoIndex");
        setupEpg(_epgMap, et -> ((VendorLargeData)et).getStringUniqueIndex(), (et, vl) -> ((VendorLargeData)et).setStringUniqueIndex((String)vl), "stringUniqueIndex");
        setupEpg(_epgMap, et -> ((VendorLargeData)et).getIntflgIndex(), (et, vl) -> ((VendorLargeData)et).setIntflgIndex(cti(vl)), "intflgIndex");
        setupEpg(_epgMap, et -> ((VendorLargeData)et).getNumericIntegerIndex(), (et, vl) -> ((VendorLargeData)et).setNumericIntegerIndex(cti(vl)), "numericIntegerIndex");
        setupEpg(_epgMap, et -> ((VendorLargeData)et).getNumericIntegerNoIndex(), (et, vl) -> ((VendorLargeData)et).setNumericIntegerNoIndex(cti(vl)), "numericIntegerNoIndex");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "vendor_large_data";
    protected final String _tableDispName = "vendor_large_data";
    protected final String _tablePropertyName = "vendorLargeData";
    protected final TableSqlName _tableSqlName = new TableSqlName("vendor_large_data", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnLargeDataId = cci("LARGE_DATA_ID", "LARGE_DATA_ID", null, null, Long.class, "largeDataId", null, true, false, true, "BIGINT", 19, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnStringIndex = cci("STRING_INDEX", "STRING_INDEX", null, null, String.class, "stringIndex", null, false, false, true, "VARCHAR", 180, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnStringNoIndex = cci("STRING_NO_INDEX", "STRING_NO_INDEX", null, null, String.class, "stringNoIndex", null, false, false, true, "VARCHAR", 180, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnStringUniqueIndex = cci("STRING_UNIQUE_INDEX", "STRING_UNIQUE_INDEX", null, null, String.class, "stringUniqueIndex", null, false, false, true, "VARCHAR", 180, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnIntflgIndex = cci("INTFLG_INDEX", "INTFLG_INDEX", null, null, Integer.class, "intflgIndex", null, false, false, true, "INT", 10, 0, null, false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnNumericIntegerIndex = cci("NUMERIC_INTEGER_INDEX", "NUMERIC_INTEGER_INDEX", null, null, Integer.class, "numericIntegerIndex", null, false, false, true, "DECIMAL", 8, 0, null, false, null, null, null, null, null, false);
//"df.Numeric"
    protected final ColumnInfo _columnNumericIntegerNoIndex = cci("NUMERIC_INTEGER_NO_INDEX", "NUMERIC_INTEGER_NO_INDEX", null, null, Integer.class, "numericIntegerNoIndex", null, false, false, true, "DECIMAL", 8, 0, null, false, null, null, null, null, null, false);
//"df.Numeric"

    /**
     * LARGE_DATA_ID: {PK, NotNull, BIGINT(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnLargeDataId() { return _columnLargeDataId; }
    /**
     * STRING_INDEX: {NotNull, VARCHAR(180)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnStringIndex() { return _columnStringIndex; }
    /**
     * STRING_NO_INDEX: {NotNull, VARCHAR(180)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnStringNoIndex() { return _columnStringNoIndex; }
    /**
     * STRING_UNIQUE_INDEX: {UQ, NotNull, VARCHAR(180)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnStringUniqueIndex() { return _columnStringUniqueIndex; }
    /**
     * INTFLG_INDEX: {NotNull, INT(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnIntflgIndex() { return _columnIntflgIndex; }
    /**
     * NUMERIC_INTEGER_INDEX: {NotNull, DECIMAL(8)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNumericIntegerIndex() { return _columnNumericIntegerIndex; }
    /**
     * NUMERIC_INTEGER_NO_INDEX: {NotNull, DECIMAL(8)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnNumericIntegerNoIndex() { return _columnNumericIntegerNoIndex; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnLargeDataId());
        ls.add(columnStringIndex());
        ls.add(columnStringNoIndex());
        ls.add(columnStringUniqueIndex());
        ls.add(columnIntflgIndex());
        ls.add(columnNumericIntegerIndex());
        ls.add(columnNumericIntegerNoIndex());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnLargeDataId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // -----------------------------------------------------
    //                                        Unique Element
    //                                        --------------
    public UniqueInfo uniqueOf() { return hpcui(columnStringUniqueIndex()); }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.VendorLargeData"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.VendorLargeDataCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.VendorLargeDataBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<VendorLargeData> getEntityType() { return VendorLargeData.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public VendorLargeData newEntity() { return new VendorLargeData(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((VendorLargeData)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((VendorLargeData)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}

File: src/dbflute/adf/meta/vendorLargeDatadbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type VendorLargeDataDbm_T struct {
	df.BaseDBMeta
	ColumnLargeDataId *df.ColumnInfo
	ColumnStringIndex *df.ColumnInfo
	ColumnStringNoIndex *df.ColumnInfo
	ColumnStringUniqueIndex *df.ColumnInfo
	ColumnIntflgIndex *df.ColumnInfo
	ColumnNumericIntegerIndex *df.ColumnInfo
	ColumnNumericIntegerNoIndex *df.ColumnInfo
}

func (b *VendorLargeDataDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *VendorLargeDataDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var VendorLargeDataDbm *VendorLargeDataDbm_T

func Create_VendorLargeDataDbm() {
	VendorLargeDataDbm = new(VendorLargeDataDbm_T)
	VendorLargeDataDbm.TableDbName = "vendor_large_data"
	VendorLargeDataDbm.TableDispName = "vendor_large_data"
	VendorLargeDataDbm.TablePropertyName = "vendorLargeData"
	VendorLargeDataDbm.TableSqlName = new(df.TableSqlName)
	VendorLargeDataDbm.TableSqlName.TableSqlName = "vendor_large_data"
	VendorLargeDataDbm.TableSqlName.CorrespondingDbName = VendorLargeDataDbm.TableDbName

	var vendorLargeData df.DBMeta
	vendorLargeData = VendorLargeDataDbm
	VendorLargeDataDbm.DBMeta=&vendorLargeData
	largeDataIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo LARGE_DATA_ID
	largeDataIdSqlName.ColumnSqlName = "LARGE_DATA_ID"
	largeDataIdSqlName.IrregularChar = false
	VendorLargeDataDbm.ColumnLargeDataId = df.CCI(&vendorLargeData, "LARGE_DATA_ID", largeDataIdSqlName, "", "", "Long.class", "largeDataId", "", true, false,true, "BIGINT", 19, 0, "",false,"","", "","","",false,"int64")
	stringIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo STRING_INDEX
	stringIndexSqlName.ColumnSqlName = "STRING_INDEX"
	stringIndexSqlName.IrregularChar = false
	VendorLargeDataDbm.ColumnStringIndex = df.CCI(&vendorLargeData, "STRING_INDEX", stringIndexSqlName, "", "", "String.class", "stringIndex", "", false, false,true, "VARCHAR", 180, 0, "",false,"","", "","","",false,"string")
	stringNoIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo STRING_NO_INDEX
	stringNoIndexSqlName.ColumnSqlName = "STRING_NO_INDEX"
	stringNoIndexSqlName.IrregularChar = false
	VendorLargeDataDbm.ColumnStringNoIndex = df.CCI(&vendorLargeData, "STRING_NO_INDEX", stringNoIndexSqlName, "", "", "String.class", "stringNoIndex", "", false, false,true, "VARCHAR", 180, 0, "",false,"","", "","","",false,"string")
	stringUniqueIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo STRING_UNIQUE_INDEX
	stringUniqueIndexSqlName.ColumnSqlName = "STRING_UNIQUE_INDEX"
	stringUniqueIndexSqlName.IrregularChar = false
	VendorLargeDataDbm.ColumnStringUniqueIndex = df.CCI(&vendorLargeData, "STRING_UNIQUE_INDEX", stringUniqueIndexSqlName, "", "", "String.class", "stringUniqueIndex", "", false, false,true, "VARCHAR", 180, 0, "",false,"","", "","","",false,"string")
	intflgIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo INTFLG_INDEX
	intflgIndexSqlName.ColumnSqlName = "INTFLG_INDEX"
	intflgIndexSqlName.IrregularChar = false
	VendorLargeDataDbm.ColumnIntflgIndex = df.CCI(&vendorLargeData, "INTFLG_INDEX", intflgIndexSqlName, "", "", "Integer.class", "intflgIndex", "", false, false,true, "INT", 10, 0, "",false,"","", "","","",false,"int64")
	numericIntegerIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo NUMERIC_INTEGER_INDEX
	numericIntegerIndexSqlName.ColumnSqlName = "NUMERIC_INTEGER_INDEX"
	numericIntegerIndexSqlName.IrregularChar = false
	VendorLargeDataDbm.ColumnNumericIntegerIndex = df.CCI(&vendorLargeData, "NUMERIC_INTEGER_INDEX", numericIntegerIndexSqlName, "", "", "Integer.class", "numericIntegerIndex", "", false, false,true, "DECIMAL", 8, 0, "",false,"","", "","","",false,"df.Numeric")
	numericIntegerNoIndexSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo NUMERIC_INTEGER_NO_INDEX
	numericIntegerNoIndexSqlName.ColumnSqlName = "NUMERIC_INTEGER_NO_INDEX"
	numericIntegerNoIndexSqlName.IrregularChar = false
	VendorLargeDataDbm.ColumnNumericIntegerNoIndex = df.CCI(&vendorLargeData, "NUMERIC_INTEGER_NO_INDEX", numericIntegerNoIndexSqlName, "", "", "Integer.class", "numericIntegerNoIndex", "", false, false,true, "DECIMAL", 8, 0, "",false,"","", "","","",false,"df.Numeric")

	VendorLargeDataDbm.ColumnInfoList = new(df.List)
	VendorLargeDataDbm.ColumnInfoList.Add(VendorLargeDataDbm.ColumnLargeDataId)
	VendorLargeDataDbm.ColumnInfoList.Add(VendorLargeDataDbm.ColumnStringIndex)
	VendorLargeDataDbm.ColumnInfoList.Add(VendorLargeDataDbm.ColumnStringNoIndex)
	VendorLargeDataDbm.ColumnInfoList.Add(VendorLargeDataDbm.ColumnStringUniqueIndex)
	VendorLargeDataDbm.ColumnInfoList.Add(VendorLargeDataDbm.ColumnIntflgIndex)
	VendorLargeDataDbm.ColumnInfoList.Add(VendorLargeDataDbm.ColumnNumericIntegerIndex)
	VendorLargeDataDbm.ColumnInfoList.Add(VendorLargeDataDbm.ColumnNumericIntegerNoIndex)


	VendorLargeDataDbm.ColumnInfoMap=make(map[string]int)
	VendorLargeDataDbm.ColumnInfoMap["largeDataId"]=0
		VendorLargeDataDbm.ColumnInfoMap["stringIndex"]=1
		VendorLargeDataDbm.ColumnInfoMap["stringNoIndex"]=2
		VendorLargeDataDbm.ColumnInfoMap["stringUniqueIndex"]=3
		VendorLargeDataDbm.ColumnInfoMap["intflgIndex"]=4
		VendorLargeDataDbm.ColumnInfoMap["numericIntegerIndex"]=5
		VendorLargeDataDbm.ColumnInfoMap["numericIntegerNoIndex"]=6
	    VendorLargeDataDbm.PrimaryKey = true
    VendorLargeDataDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &vendorLargeData
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(VendorLargeDataDbm.ColumnLargeDataId)

	VendorLargeDataDbm.PrimaryInfo = new(df.PrimaryInfo)
	VendorLargeDataDbm.PrimaryInfo.UniqueInfo = ui
	
	var vendorLargeDataMeta df.DBMeta = VendorLargeDataDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["VendorLargeData"] = &vendorLargeDataMeta
}
